src/System.Security.Cryptography.Pkcs/tests/Pkcs9AttributeTests.cs

```diff
@@ -28,16 +28,30 @@ public static void Pkcs9AttributeAsnEncodedDataCtorNullOid()
         }
 
         [Fact]
-        public static void TestDateBefore1950()
+        public static void TestDateBefore1950_Utc()
         {
-            DateTime dt = new DateTime(1949, 12, 31, 11, 59, 59);
+            DateTime dt = new DateTime(1949, 12, 31, 11, 59, 59, DateTimeKind.Utc);
             Assert.ThrowsAny<CryptographicException>(() => new Pkcs9SigningTime(dt));
         }
 
         [Fact]
-        public static void TestDateAfter1949()
+        public static void TestDateBefore1950_Local()
         {
-            DateTime dt = new DateTime(2050, 01, 01, 00, 00, 00);
+            DateTime dt = new DateTime(1949, 12, 31, 11, 59, 59, DateTimeKind.Local);
+            Assert.ThrowsAny<CryptographicException>(() => new Pkcs9SigningTime(dt));
```

Comment:
This test passes for me on .NET Framework (PDT / UTC-7)

```
12/31/1949 11:59:59 PM (Unspecified)
CryptographicException: An error occurred during encode or decode operation.
12/31/1949 11:59:59 PM (Local)
Success
12/31/1949 11:59:59 PM (Utc)
CryptographicException: An error occurred during encode or decode operation.
1/1/1950 12:00:00 AM (Unspecified)
Success
1/1/1950 12:00:00 AM (Local)
Success
1/1/1950 12:00:00 AM (Utc)
Success
12/31/2049 11:59:59 PM (Unspecified)
Success
12/31/2049 11:59:59 PM (Local)
CryptographicException: An error occurred during encode or decode operation.
12/31/2049 11:59:59 PM (Utc)
Success
1/1/2050 12:00:00 AM (Unspecified)
CryptographicException: An error occurred during encode or decode operation.
1/1/2050 12:00:00 AM (Local)
CryptographicException: An error occurred during encode or decode operation.
1/1/2050 12:00:00 AM (Utc)
CryptographicException: An error occurred during encode or decode operation.
```

I assume it's passing in CI because the machines are set to UTC.

A better way seems to be to take the UTC minimum time, convert it to local, subtract a second, and then use that.

Similar (but opposite) for the maximum time, just so that it doesn't make assumptions about the test user's timezone.


```C#
DateTime[] times =
{
    new DateTime(1949, 12, 31, 23, 59, 59),
    new DateTime(1949, 12, 31, 23, 59, 59, DateTimeKind.Local),
    new DateTime(1949, 12, 31, 23, 59, 59, DateTimeKind.Utc),
    new DateTime(1950, 1, 1, 0, 0, 0),
    new DateTime(1950, 1, 1, 0, 0, 0, DateTimeKind.Local),
    new DateTime(1950, 1, 1, 0, 0, 0, DateTimeKind.Utc),
    new DateTime(2049, 12, 31, 23, 59, 59),
    new DateTime(2049, 12, 31, 23, 59, 59, DateTimeKind.Local),
    new DateTime(2049, 12, 31, 23, 59, 59, DateTimeKind.Utc),
    new DateTime(2050, 1, 1, 0, 0, 0),
    new DateTime(2050, 1, 1, 0, 0, 0, DateTimeKind.Local),
    new DateTime(2050, 1, 1, 0, 0, 0, DateTimeKind.Utc),
};

foreach (DateTime dt in times)
{
    Console.WriteLine($"{dt} ({dt.Kind})");

    try
    {
        Pkcs9SigningTime st = new Pkcs9SigningTime(dt);
        Console.WriteLine("Success");
    }
    catch (CryptographicException ce)
    {
        Console.WriteLine($"CryptographicException: {ce.Message}");
    }
}
```